app/controllers/hikes_controller.rb

```diff
@@ -0,0 +1,17 @@
+class HikesController < ApplicationController
+
+  def index
+    hikes = Hike.all
+
+    render json: hikes.as_json
+  end
+
+  def show
+    hike_id = params["id"]
+    hike = Hike.find_by(id: hike_id)
+
+    render json: hike.as_json 
+
+  end
+
```

Comment:
it's a good practice to use "strong parameters" in your controllers where you "allow-list" parameter values 

```ruby
  def show_params
    params.require(:id)
  end
```

then in your controller method you use it...

```ruby
  def show
    hike = Hike.find(show_params[:id])

    render json: hike.as_json 
  end
```

If you use strong params and enforce what the controller expects to receive, you can get really good error messaging built in when a call isn't working, like maybe you're not building the request right in the front end.

Reply:
updated with strong parameters.